iree/compiler/Conversion/LinalgToLLVM/HALInterfaceToMemrefArguments.cpp

```diff
@@ -143,10 +153,27 @@ struct ProcessFuncInterfacePattern : public OpConversionPattern<FuncOp> {
     for (auto bufferOp : bufferOps) {
       bufferOp.replaceAllUsesWith(
           newFuncOp.getArgument(bufferArgMap[bufferOp]));
+
       rewriter.eraseOp(bufferOp);
     }
 
     rewriter.eraseOp(funcOp);
+    Type indexType = rewriter.getIndexType();
```

Comment:
Nit: typically we want to keep `rewriter.eraseOp` at the very last before returning success. It's easier to find out that way.